server/httpin/restapi/call.go

```diff
@@ -28,91 +26,26 @@ func (a *restapi) initCall(h *httpin.Handler) {
 func (a *restapi) Call(r *incoming.Request, w http.ResponseWriter, req *http.Request) {
 	creq, err := apps.CallRequestFromJSONReader(req.Body)
 	if err != nil {
-		httputils.WriteError(w, utils.NewInvalidError(errors.Wrap(err, "failed to unmarshal Call request")))
+		err = errors.Wrap(err, "failed to unmarshal Call request")
+		r.Log.WithError(err).Infof("incoming call failed")
+		httputils.WriteError(w, utils.NewInvalidError(err))
 		return
 	}
-	r.SetAppID(creq.Context.AppID)
-
-	// Clear out anythging in the incoming expanded context for security
-	// reasons, it will be set by Expand before passing to the app.
-	creq.Context.ExpandedContext = apps.ExpandedContext{}
-	creq.Context, err = a.cleanUserAgentContext(r.ActingUserID(), creq.Context)
-	if err != nil {
-		httputils.WriteError(w, utils.NewInvalidError(errors.Wrap(err, "invalid call context for user")))
+	if creq.Context.UserAgentContext.AppID == "" {
+		err = errors.New("app ID is not set in Call request")
+		r.Log.WithError(err).Infof("incoming call failed")
+		httputils.WriteError(w, utils.NewInvalidError(err))
 		return
 	}
+	r = r.WithDestination(creq.Context.UserAgentContext.AppID)
 
-	cresp := a.proxy.Call(r, *creq)
-
-	// Add the request and response digests to the logger.
-	r.Log = r.Log.With(creq, cresp)
+	// Call the app.
+	cresp := a.Proxy.InvokeCall(r, *creq)
 
 	// Only track submit calls.
 	if creq.Context.UserAgentContext.TrackAsSubmit {
-		a.conf.Telemetry().TrackCall(string(creq.Context.AppID), string(creq.Context.Location), creq.Context.ActingUserID, "submit")
+		a.Config.Telemetry().TrackCall(string(creq.Context.AppID), string(creq.Context.Location), r.ActingUserID(), "submit")
```

Comment:
1/5 only log successes, aggregate errors somehow? problem: there's no longer a pre-check until Expand, so there may be a lot of access violation errors. 

1/5 move the telemetry submissions into the `proxy` and `appservices` packages, out of `httpin`? cc @mickmister